src/linear_samplers/block_row_sample_wo_replacement.jl

```diff
@@ -0,0 +1,104 @@
+- `population::Union{Vector{Int64}, Nothing}`, buffer array to hold `collect(1:size(A)[1])` used in `sample`.
+- `rows_sampled::Union{Vector{Int64}, Nothing}`, buffer array to hold index of rows sampled from `A`.
+- `S::Union{Matrix{Int64}, Nothing}`, buffer array to hold sketched matrix `S`.
+
+Calling `LinSysBlkRowSelectWoReplacement()` defaults to `LinSysBlkRowSelectWoReplacement(2, nothing, nothing, nothing, nothing)`.
+
+An additional constructor is provided with keyword arguments `block_size` and `probability`.
+"""
+mutable struct LinSysBlkRowSelectWoReplacement <: LinSysVecRowSelect
+    block_size::Int64
+    probability::Union{Weights, Vector{Float64}, Nothing}
+    population::Union{Vector{Int64}, Nothing}
+    rows_sampled::Union{Vector{Int64}, Nothing}
+    S::Union{Matrix{Int64}, Nothing}
+end
+
+function LinSysBlkRowSelectWoReplacement(;block_size=2, probability=nothing)
+    return LinSysBlkRowSelectWoReplacement(block_size, probability, nothing, nothing, nothing)
+end
+
+# Common sample interface for linear systems
+function sample(
+    type::LinSysBlkRowSelectWoReplacement,
+    A::AbstractArray,
+    b::AbstractVector,
+    x::AbstractVector,
+    iter::Int64
+)
+
+    # Error checking and initialization on first iteration
+    if iter == 1
+        nrow = size(A)[1]
+        if type.block_size <= 0
+            throw(DomainError("block_size is 0 or negative!")) 
+        end
+        
+        if type.block_size > nrow
+            throw(DomainError("block_size is larger than the number of rows in A! Cannot select this amount without replacement!"))
+        end
+
+        # initialization buffer arrays
+        type.S = Matrix{Int64}(undef, type.block_size, nrow)
+        type.rows_sampled = Vector{Int64}(undef, type.block_size)
+
+        # error checking on probability vector, possible initialization and type conversion
+        if isnothing(type.probability)
+            type.probability = Weights(repeat([1/nrow], outer = nrow))
+        elseif isa(type.probability, Vector) || isa(type.probability, Weights)
+            # check that probability is a valid distribution on the rows 
+            if !(sum(type.probability) ≈ 1)
+                throw(DomainError("Elements of probability do not sum to 1!"))
+            elseif sum(type.probability .>= 0) != size(type.probability)[1]
+                throw(DomainError("Not all probabilities are non-negative in probability!")) 
+            elseif size(type.probability)[1] != size(A)[1]
+                throw(DimensionMismatch("Length of probability vector is smaller than the number of rows in A!"))
+            elseif sum(type.probability .> 0) < type.block_size
+                throw(DimensionMismatch("Not enough non-zero probabilities in probability to select the required number of rows!"))
```

Comment:
Really like the specificity of these warnings maybe place probability in '' to emphasize you are referring to the variable.